### PR TITLE
Update release-checklist.md

### PR DESCRIPTION
**Pull request by JLDLaughlin**

### CODE REVIEW

File: doc/developer/release-checklist.md [hunk 1]
```diff
@@ -36,9 +41,11 @@
   ```shell
   tag=v<VERSION>-rc<N>
   git tag -a $tag -m $tag
-  git push origin $tag
+  git push upstream $tag
```
Comment (quodlibetor) [inline doc/developer/release-checklist.md]: I think we should just document that `origin` should be the `MaterializeInc/materialize` remote, I think our default instructions leave `origin` as that.

I think I'd change this to something like:

```
git push origin $tag  # where 'origin' is your MaterializeInc/materialize remote
```

File: doc/developer/release-checklist.md [hunk 2]
```diff
@@ -53,6 +60,23 @@
...
+    ```shell script
+    level=error ts=2020-03-13T18:29:51.472Z caller=query_logger.go:82 component=activeQueryTracker msg="Error opening query log file" file=/prometheus/queries.active err="open /prometheus/queries.active: permission denied"
+    panic: Unable to create mmap-ed active query log
+    ```
+
+    Do the following:
+    ```shell script
+    sudo chmod 777 prometheus/data
+    ./dc.sh up prometheus
+    ```
```
Comment (quodlibetor) [inline doc/developer/release-checklist.md]: I think we could just add `chmod 777 prometheus/data` to dc.sh and have this work reliably for everyone?

File: doc/developer/release-checklist.md [hunk 3]
```diff
@@ -82,7 +115,7 @@
   ```shell
   tag=v<VERSION>
   git tag -a $tag -m $tag
-  git push origin $tag
+  git push upstream $tag
```
Comment (benesch) [inline doc/developer/release-checklist.md]: (Ditto above, just in case.)

File: doc/developer/release-checklist.md (hunk 1 above)
Comment (JLDLaughlin) [reply doc/developer/release-checklist.md]: Okay! Will fix.

Personally, I've had `upstream` as my MaterializeInc/materialize remote and `origin` as my fork. I can't seem to find where we tell people that `origin` should point to Materialize, do you know where it is?

Comment (quodlibetor) [reply doc/developer/release-checklist.md]: > Personally, I've had upstream as my MaterializeInc/materialize remote
that might have happened during the great publification? 

When you `git clone URL` a repo that url is default marked as the `origin` remote, and then `git remote add NAME URL` marks that one as `NAME`, so depending on the order of clones you might have needed to come up with another name. `upstream` also sounds like something that I have called the same kind of remote that you're calling it, so I might have advised you to do this.

> Where we tell people that origin should point to Materialize

It's not explicit, but this clones `MaterializeInc/materialize`, although we should probably add another step that gets people to fork it and add the remote: https://github.com/MaterializeInc/mtrlz-setup/blob/794e513017/mac/materialize.sh#L88

Comment (JLDLaughlin) [reply doc/developer/release-checklist.md]: Ah got it, thank you!

**Pull request merged**